src/MapEditorContext/MapEditorContextProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+    createContext,
+    Dispatch,
+    FunctionComponent,
+    ReactNode,
+    useReducer
+} from "react";
+import { ActiveItem } from "../models/ActiveItem";
+import { Platform } from "../models/Platform";
+import { Vector2 } from "../models/Vector2";
+import { MapEditorContextActions } from "./MapEditorContextActions";
+import { mapEditorContextReducer } from "./MapEditorContextReducer";
+
+export type MapEditorContextState = {
+    width: number;
+    height: number;
+    name: string;
+    activeItem?: ActiveItem;
+    mousePosition: Vector2;
+    grid: number;
+    platforms: Platform[];
+};
+
+const initialMapEditorContext: MapEditorContextState = {
+    width: 1280,
+    height: 720,
+    name: "New Map",
+    mousePosition: [0, 0],
+    grid: 8,
+    platforms: []
+};
+
+export const MapEditorContext = createContext<MapEditorContextState | null>(
```

Comment:
Do you like using a `Nullable` type?

```
type Nullable<T> =  T | null;
```

You'd end up with

```suggestion
export const MapEditorContext = createContext<Nullable<MapEditorContextState>>(
```

Reply:
I think that's a good idea